public/pages/workflow_detail/components/header.tsx

```diff
@@ -52,14 +73,130 @@ export function WorkflowDetailHeader(props: WorkflowDetailHeaderProps) {
     }
   }, [props.workflow]);
 
-  return (
+  const { TopNavMenu, HeaderControl } = getNavigationUI();
+  const { setAppRightControls } = getApplication();
+  const {
+    chrome: { setHeaderVariant },
+  } = getCore();
+  const uiSettings = getUISettings();
+  const showActionsInHeader = uiSettings.get('home:useNewHomePage');
+
+  useEffect(() => {
+    if (showActionsInHeader) {
+      setHeaderVariant?.(HeaderVariant.APPLICATION);
+    }
+
+    return () => {
+      setHeaderVariant?.();
+    };
+  }, [setHeaderVariant, showActionsInHeader]);
+
+  const onExportButtonClick = () => {
+    setIsExportModalOpen(true);
+  };
+
+  const onExitButtonClick = () => {
+    history.replace(
+      `${APP_PATH.WORKFLOWS}${
+        dataSourceId !== undefined ? `?dataSourceId=${dataSourceId}` : ''
+      }`
+    );
+  };
+
+  const topNavConfig: TopNavMenuData[] = [
+    {
+      iconType: 'exportAction',
+      tooltip: 'Export',
+      ariaLabel: 'Export',
+      run: onExportButtonClick,
+      controlType: 'icon',
+    } as TopNavMenuIconData,
+    {
+      iconType: 'exit',
+      tooltip: 'Return to projects',
+      ariaLabel: 'Exit',
+      run: onExitButtonClick,
+      controlType: 'icon',
+    } as TopNavMenuIconData,
+  ];
+
+  const dataSourceEnabled = getDataSourceEnabled().enabled;
+  const dataSourceId = getDataSourceId();
+
+  let renderDataSourceComponent: ReactElement | null = null;
+  if (dataSourceEnabled && getDataSourceManagementPlugin()) {
+    const DataSourceMenu = getDataSourceManagementPlugin().ui.getDataSourceMenu<
+      DataSourceViewConfig
+    >();
+    renderDataSourceComponent = (
+      <DataSourceMenu
+        setMenuMountPoint={props.setActionMenu}
+        componentType={'DataSourceView'}
+        componentConfig={{
+          activeOption: [{ id: dataSourceId }],
+          fullWidth: false,
+          savedObjects: getSavedObjectsClient(),
+          notifications: getNotifications(),
+        }}
+      />
+    );
+  }
+
+  return showActionsInHeader ? (
+    <>
+      {isExportModalOpen && (
+        <ExportModal
+          workflow={props.workflow}
+          setIsExportModalOpen={setIsExportModalOpen}
+        />
+      )}
```

Comment:
duplicate of lines 193-198. Since this state isn't dependent on `showActionsInHeader` being true or false, let's defined this outside of that first. Something like:
```
return ( 
<>
{ isExportModalOpen && (...) }
{ showActionsInHeader ? (<top nav menu stuff>) : (<existing header stuff>) }
</>
```